.github/workflows/ci-run-test.yml

```diff
@@ -229,8 +229,8 @@ jobs:
       - name: Install Python deps
         run: |
           python -m pip install --upgrade pip requests wheel urllib3 chardet build
-      - name: Install poetry
-        run: pipx install poetry
```

Comment:
Because pipx is part of the base install of the image, it's using the base python. That in turn makes poetry use that version of Python for venvs. By using pip (which will be setup by the setup Python action) we ensure that poetry uses the Python version we're testing.